OpenAPI/OpenAPI.NET/references-openapi.md

```diff
@@ -0,0 +1,665 @@
+                    $id: 'http://example.org/c'
+                    type:
+                        - object
+                        - 'null'
+                    additionalProperties: false
+                    properties:
+                        b:
+                            $ref: '#/properties/d'
+                        d:
+                            type: string
+```
+
+### What kinds of JSON Schema references exist
+
+JSON Schema references can use either a locator or an identifier. Locators indicate where to find the target schema based on the name and structure of documents.  However, identifiers are opaque URIs that match to the identifer of a JSON schema that has a $id either explicitly or implicitly by inheriting an identity from its parent.
+
+### What can a JSON Schema locator reference target
+
+- An internal component
+- An interal component subschema
+- An internal inline subschema
+- An external OpenApi document component
+- An external inline subchema [Not supported]
+- An external inline subchema using anchor
+- An external fragment [Not supported]
+
+#### An internal component
+
+```yaml
+openapi: 3.1.0
+info:
+    title: Reference to an internal component
+    version: 1.0.0
+paths:
+    /item:
+        get:
+          responses:
+            200:
+              description: ok
+              content:
+                application/json:
+                  schema:
+                    $ref: '#/components/schemas/item'
+components:
+  schemas:
+      item:
+        type: object
+```
+
+#### An internal component subschema
+
+```yaml
+openapi: 3.1.0
+info:
+    title: Reference to an internal component
+    version: 1.0.0
+paths:
+    /person/{id}:
+        get:
+          responses:
+            200:
+              description: ok
+              content:
+                application/json:
+                  schema:
+                    $ref: '#/components/schemas/person'
+    /person/{id}/address:
+        get:
+          responses:
+            200:
+              description: ok
+              content:
+                application/json:
+                  schema:
+                    $ref: '#/components/schemas/person/properties/address'
+components:
+  schemas:
+      person:
+        type: object
+        properties:
+            name:
+                type: string
+            address:
+                type: object
+                properties:
+                    street:
+                        type: string
+                    city:
+                        type: string
+```
+
+
+#### An external OpenApi document component
+
+```yaml
+openapi: 3.1.0
+info:
+    title: Reference to an external OpenApi document component
+    version: 1.0.0
+paths:
+    /person/{id}:
+        get:
+          responses:
+            200:
+              description: ok
+              content:
+                application/json:
+                  schema:
+                    $ref: 'OAS-schemas.yaml#/components/schemas/person'
```

Comment:
in terms of URI here, what conventions can be applied?
is this equivalent to having `./` in front? What about `/` in front? Do we have an RFC we can leverage here?